### DISCUSSION TITLE
Can we get an offical FlatLaf dark theme port for avalonia ui?

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [AvaloniaUI/Avalonia#15985] [Ideas]
### Is your feature request related to a problem? Please describe.

I don't like look of any existing theme for avalonia ui and would prefer a more traditional desktop theme 

### Describe the solution you'd like

Can we get a official support for a clone of java swing's flatlaf dark theme?
https://www.formdev.com/flatlaf/

### Describe alternatives you've considered

I considered imgui.net but it tends to crash alot and have odd issues.
I have also considered moving to java and using swing itself.

### Additional context

I am looking for a cross platform C# desktop gui I can use to build a game engine on.

**Comment by stevemonaco** [comment]
Creating an entirely new theme for every control and control state can take well over a hundred hours for initial release without counting for maintenance / bug fixing. You can look through some third-party themes here: https://github.com/AvaloniaCommunity/awesome-avalonia?tab=readme-ov-file#theme--icons

I personally started with Simple theme and tweaked colors + spacing for a flat / compact look and it's much less work to style a smaller subset of components.

I can't speak for the team, but I would expect another desktop theme to take a backseat to a Material3/Cupertino port geared towards mobile if there were available time to create another official theme.

**Comment by Shadowblitz16** [comment]
> I can't speak for the team, but I would expect another desktop theme to take a backseat to a Material3/Cupertino port geared towards mobile if there were available time to create another official theme.

There is already enough mobile themes though, most of .NET gui frameworks are nothing but mobile/hybrid stuff.
We really need more classic desktop oriented stuff.

**Reply by thevortexcloud** [reply]
How classic are we talking?

https://github.com/Splitwirez/avalonia-aero-theme

Anyway, what's the problem with the simple/fluent themes?

**Reply by stevemonaco** [reply]
I assume they want something really dense, flat, and plain like Imgui is. Avalonia does lack a bit for "power user" apps out-of-the-box, but some customization is necessary for every app. It's not too hard to increase the density though Fluent has too much spacing for this use case, even with `Compact` density. I do strongly disagree with the "nothing but mobile/hybrid" claim though as .NET has basically no pure mobile themes / look-and-feels other than .NET Maui's native control wrappers.

**Reply by thevortexcloud** [reply]
Assuming they don't need every control: An imgui style theme could probably be done in a day or so by just overriding one of the default themes as needed. A proper generic theme would take longer though. 

That said: I also would like some more official theme options. I also don't expect any to be added any time soon either. Also most classic desktop design has been on the way out for awhile now due to it being terrible for most touch screen devices.

**Reply by thevortexcloud** [reply]
Someone made an even more classic theme:

https://github.com/BAndysc/Classic.Avalonia

**Comment by timunie** [comment]
Check out awesome avalonia for other available themes. 

Tbh, I don't see such a theme provided soonish if even from our side.

**Comment by Shadowblitz16** [comment]
> Check out awesome avalonia for other available themes.
> 
> Tbh, I don't see such a theme provided soonish if even from our side.

I already did before I made this issue.

**Comment by danipen** [comment]
Did you check https://github.com/Actipro/Avalonia-Controls?

**Reply by Shadowblitz16** [reply]
> Did you check https://github.com/Actipro/Avalonia-Controls?

No it still looks like a web/mobile app not a classic desktop app

**Reply by stevemonaco** [reply]
Can you post a screenshot of your idea of a "classic desktop app" because this is how UX has been for nearly a decade. If the Avalonia Simple theme is not sufficient, then we're going back further than 15-20 years. FlatLaf really isn't *that* different, it's mainly a bit more dense which you can rework well enough in most themes.

**Reply by Shadowblitz16** [reply]
> Can you post a screenshot of your idea of a "classic desktop app" because this is how UX has been for nearly a decade. If the Avalonia Simple theme is not sufficient, then we're going back further than 15-20 years. FlatLaf really isn't _that_ different, it's mainly a bit more dense which you can rework well enough in most themes.

I am talking about thin controls that don't take up space rather then fat controls that would be useful on mobile, visual controls like groupboxes, inset and outset panels and window / dialog oriented gui rather then page oriented.

**Reply by thevortexcloud** [reply]
`HeaderedContentControl` does exist which can be used like a group box. Anyway, it sounds like you want a Windows 7 era theme. Which I am not sure is a good idea even if you only target desktop platforms. So many computers now have touchscreens, and people do use them. EG laptops that can be folded into a tablet are pretty popular.

**Reply by Shadowblitz16** [reply]
What does GritGene and Unity use as it's theme?

**Reply by thevortexcloud** [reply]
Unity does not use Avalonia for its UI.  I think it's some custom thing and the theme is made by them as far as I am aware. No idea what GritGene is.

**Comment by rabbitism** [comment]
I will take a look at it.

**Reply by Shadowblitz16** [reply]
@rabbitism how is it going?
Did you see how hard it would be to make a flatlaf recreation?

**Comment by Shadowblitz16** [comment]
This is still an issue both simple and fluent themes are too spacious
I mean NumericUpAndDown in fluent has massive up and down buttons to the point they wrap around on one like and that's with Compact Version.

And NumericUpAndDown in simple has way too much padding around the text.

I am looking for a more traditional style.
Compact and made for desktop, not mobile. 

I don't want to have to theme lookless controls, I just want something I can use that looks good on desktop across platforms.

**Reply by thevortexcloud** [reply]
There is nothing stopping you from modifying the existing themes. You can either use it as a base to make a custom theme, or override them with styles.

**Reply by Shadowblitz16** [reply]
> There is nothing stopping you from modifying the existing themes. You can either use it as a base to make a custom theme, or override them with styles.

I have tried the theming system is just too complex to understand for me.

**Comment by Shadowblitz16** [comment]
There is the grit world theme that looks really nice that you guys showcased why don't we have something like that instead of the awful ones we have now?

**Reply by MikeCodesDotNET** [reply]
That was made by them. Just because we've including an app in the showcase, doesn't mean its source will be available.
